magento1-module/app/code/local/Divante/VueStorefrontBridge/controllers/UserController.php

```diff
@@ -321,29 +323,61 @@ public function meAction(){
                         foreach($updatedCustomer['addresses'] as $updatedAdress) {
                             $updatedAdress['region'] = $updatedAdress['region']['region'];
 
-                            if($updatedAdress['default_billing']) {
+                            if ($updatedAdress['delete'] === true) {
+                                try {
+                                    $sAddress = Mage::getModel('customer/address');
+
+                                    $sAddress->load($updatedAdress['entity_id']);
+
+                                    $sAddress->setData($updatedAdress)->delete();
+                                } catch (Throwable $exception) {
+                                    Mage::log(
+                                        'Cloud not load address:' . $updatedAdress['entity_id'] . " to delete it",
+                                        Zend_Log::ERR,
+                                        'vsbridge.log'
+                                    );
+                                }
+                            } elseif ($updatedAdress['default_billing']) {
                                 $bAddressId = $customer->getDefaultBilling();
                                 $bAddress = Mage::getModel('customer/address');
 
-                                if($bAddressId)
-                                    $bAddress->load($bAddressId);
+                                if ($bAddressId) $bAddress->load($bAddressId);
 
                                 $updatedAdress['parent_id'] = $customer->getId();
-
+                                $updatedAdress['street'] = $addressHelper->concatStreetData($updatedAdress['street']);
 
                                 $bAddress->setData($updatedAdress)->setIsDefaultBilling(1)->save();
                                 $updatedBillingId = $bAddress->getId();
-                            }
-                            if($updatedAdress['default_shipping']) {
+                            } elseif ($updatedAdress['default_shipping']) {
```

Comment:
An address can be default billing and default shipping at the same time. I' m not sure if this is allowed here since you are using an `elseif`?

Reply:
Yea you are right i need to push my latest changes here as i needed to fix a lot of logic bugs in here. 
Which are all fixed now, but i have forgotten to update this PR since then.